modules/objdetect/src/hog.cpp

```diff
@@ -3665,7 +3671,7 @@ void HOGDescriptor::detectMultiScaleROI(const cv::Mat& img,
     Mutex mtx;
 
     parallel_for_(Range(0, (int)locations.size()),
-                  HOGConfInvoker(this, img, hitThreshold, Size(8, 8),
+                  HOGConfInvoker(this, img.getMat(), hitThreshold, Size(8, 8),
```

Comment:
`.getMat()` returns temporary object which is passed by reference.
Usually it is stored by processing object as a reference too, but these temporary objects got destroyed.
Hopefully, this implementation is "not effective" and store object instead of const reference for source image - but it can be refactored and this code will become broken.

I would be better to avoid such usage and use local variable to store `cv::Mat`:
```
Mat img_ = img.getMat();
parallel_for_(...)
```